bootstrap.js

```diff
@@ -4,101 +4,200 @@ Cu.import("resource://gre/modules/Console.jsm");
+    this.removeCSS();
+
+    // Remove the copy controller
+    this.removeCopyController();
+
+    // Remove modifications to shareButton (modified in CopyController)
+    if (this.shareButton !== null) {
+      // if null this means the user did not copy from the URL bar
+      // so we don't have anything to remove
+      this.shareButton.classList.remove("social-share-button-on");
+      this.shareButton.removeEventListener("animationend", this.animationEndListener);
+    }
+  }
+}
+
+// see https://dxr.mozilla.org/mozilla-central/rev/53477d584130945864c4491632f88da437353356/xpfe/appshell/nsIWindowMediatorListener.idl
+const windowListener = {
+  onWindowTitleChange(window, title) { },
+  onOpenWindow(xulWindow) {
+    // xulWindow is of type nsIXULWindow, we want an nsIDOMWindow
+    // see https://dxr.mozilla.org/mozilla-central/rev/53477d584130945864c4491632f88da437353356/browser/base/content/test/general/browser_fullscreen-window-open.js#316
+    // for how to change XUL into DOM
+    const domWindow = xulWindow.QueryInterface(Ci.nsIInterfaceRequestor)
+      .getInterface(Ci.nsIDOMWindow);
+
+    // we need to use a listener function so that it's injected
+    // once the window is loaded / ready
+    const onWindowOpen = (e) => {
+      domWindow.removeEventListener("load", onWindowOpen);
+      const browserWindow = new BrowserWindow(domWindow);
+      browserWindow.startup();
+    };
+
+    domWindow.addEventListener("load", onWindowOpen, true);
+  },
+  onCloseWindow(window) { },
 };
 
-this.shutdown = function(data, reason) {
-  // Remove the customizationending listener
-  bWindow.removeEventListener("customizationending", insertCopyController);
+this.install = function(data, reason) {};
 
-  utils.removeSheet(CSS_URI, utils.AGENT_SHEET);
+this.startup = function(data, reason) {
+  // iterate over all open windows
+  const windowEnumerator = Services.wm.getEnumerator("navigator:browser");
+  while (windowEnumerator.hasMoreElements()) {
+    const window = windowEnumerator.getNext();
+    const browserWindow = new BrowserWindow(window);
+    browserWindow.startup();
+  }
 
-  // Remove the copy controller
-  urlInputControllers.removeController(urlInputController);
+  // add an event listener for new windows
+  Services.wm.addListener(windowListener);
+};
 
-  // if null this means the user did not copy from the URL bar
-  // so we don't have anything to remove
-  if (shareButton !== null) {
-    shareButton.classList.remove("social-share-button-on");
-    shareButton.removeEventListener("animationend", shareButtonAnimationListener);
+this.shutdown = function(data, reason) {
+  const windowEnumerator = Services.wm.getEnumerator("navigator:browser");
+  while (windowEnumerator.hasMoreElements()) {
+    const window = windowEnumerator.getNext();
+    const browserWindow = browserWindowWeakMap.get(window);
+    browserWindow.shutdown();
   }
+
+  // remove event listener for new windows
+  Services.wm.removeListener(windowListener);
```

Comment:
You should remove this listener before processing the weakmap, so that you don't have a race condition where a browser window is added during shutdown. Rare, but possible.